processing/src/main/java/io/druid/query/aggregation/LastAggregatorFactory.java

```diff
@@ -0,0 +1,335 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Doubles;
+import com.google.common.primitives.Longs;
+import com.metamx.common.IAE;
+import com.metamx.common.StringUtils;
+import io.druid.collections.SerializablePair;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.ObjectColumnSelector;
+import io.druid.segment.column.Column;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+public class LastAggregatorFactory extends AggregatorFactory
+{
+  private static final byte CACHE_TYPE_ID = 0x11;
+  private static final String TYPE_LONG = "long";
+  private static final String TYPE_DOUBLE = "double";
+
+  private final String fieldName;
+  private final String name;
+  private final String value;
+
+  @JsonCreator
+  public LastAggregatorFactory(
```

Comment:
Hmm, maybe `longLast` and `doubleLast` aren't so bad… it looks like there is really little code shared between them.